lib/mpl_toolkits/mplot3d/axis3d.py

```diff
@@ -426,57 +419,61 @@ def draw(self, renderer):
         self.offsetText.set_ha(align)
         self.offsetText.draw(renderer)
 
-        if self.axes._draw_grid and len(ticks):
-            # Grid lines go from the end of one plane through the plane
-            # intersection (at xyz0) to the end of the other plane.  The first
-            # point (0) differs along dimension index-2 and the last (2) along
-            # dimension index-1.
-            lines = np.stack([xyz0, xyz0, xyz0], axis=1)
-            lines[:, 0, index - 2] = maxmin[index - 2]
-            lines[:, 2, index - 1] = maxmin[index - 1]
-            self.gridlines.set_segments(lines)
-            self.gridlines.set_color(info['grid']['color'])
-            self.gridlines.set_linewidth(info['grid']['linewidth'])
-            self.gridlines.set_linestyle(info['grid']['linestyle'])
-            self.gridlines.do_3d_projection()
-            self.gridlines.draw(renderer)
-
-        # Draw ticks:
-        tickdir = self._get_tickdir()
-        tickdelta = deltas[tickdir]
-        if highs[tickdir]:
-            ticksign = 1
-        else:
-            ticksign = -1
+        if len(ticks):
```

Comment:
I prefer not including these kinds of checks (which lead to a large block being extra indented, and forcing the reader to keep track of when the condition stops to apply) unless it is either really necessary, or if the gain in time is really large when the block can be checked, which seems unlikely here?

Reply:
OK! I reverted that.